docs/installation.rst

```diff
@@ -1,22 +1,86 @@
-Installation
-============
+Zarr-Python 3 Migration Guide
+=============================
 
-Zarr depends on NumPy. It is generally best to `install NumPy
-<https://numpy.org/doc/stable/user/install.html>`_ first using whatever method is most
-appropriate for your operating system and Python distribution. Other dependencies should be
-installed automatically if using one of the installation methods below.
+Zarr-Python 3 introduces a number of changes to the API, including a number
+of significant breaking changes and pending deprecations.
 
-Install Zarr from PyPI::
+This page provides a guide highlighting the most notable changes to help you
+migrate your code from Zarr-Python 2.x to Zarr-Python 3.x.
 
-    $ pip install zarr
+Compatibility target
+--------------------
 
-Alternatively, install Zarr via conda::
+Zarr-Python 3 represents a major refactor of the Zarr-Python codebase. Some of the goals motivating this refactor included:
 
-    $ conda install -c conda-forge zarr
+  - adding support for the V3 specification (alongside the V3 specification)
+  - cleaning up internal and user facing APIs
+  - improving performance (particularly in high latency storage environments like cloud object store)
 
-To install the latest development version of Zarr, you can use pip with the
-latest GitHub main::
+Though these goals necessitated some breaking changes to the API (hence the major version update), we have tried to maintain
+backwards compatibility in the most widely used parts of the API including the `Array` and `Group` classes and the top-level
+API (e.g. `zarr.open_array` and `zarr.open_group`).
```

Comment:
new internal data model (2.* was tightly coupled to the zarr spec), was not possible to support the v3 spec and keep that coupling